core/src/main/scala/akka/kafka/scaladsl/MetadataClient.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka.scaladsl
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.dispatch.ExecutionContexts
+import akka.kafka.{ConsumerSettings, KafkaConsumerActor}
+import akka.kafka.Metadata.{BeginningOffsets, GetBeginningOffsets}
+import akka.pattern.ask
+import akka.util.Timeout
+import org.apache.kafka.common.TopicPartition
+
+import scala.concurrent.{ExecutionContextExecutor, Future}
+import scala.util.{Failure, Success}
+
+class MetadataClient(actorSystem: ActorSystem, timeout: Timeout) {
```

Comment:
It would be preferable to have one `MetadataClient` instance per consumer actor/settings.
Make the primary constructor `private` and let it take the consumer actor's `ActorRef`.
Create factory methods in the companion object, one taking the `ActorRef` and one taking `ConsumerSettings`.

Reply:
Thanks for this hint. I pushed new implementation.

Reply:
Thanks for the hint. I pushed new implementation.

Reply:
@ennru could you look at it again and say if you meant it?

Reply:
Yes, that looks good.
The actor should only be stopped by the metadata client if it was created by it, you'd possibly need to pass that information to the client. So two constructors on that class might make sense.
`close` might be better than `stop`.